src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -546,9 +561,33 @@ public void validate(FailureCollector failureCollector, Map<String, String> argu
         validateJobLabelKeyValue(failureCollector);
       }
 
+      if (!containsMacro(WRITE_PREFERENCE)) {
+        validateWritePreference(failureCollector, getWritePreference());
+      }
+
       failureCollector.getOrThrowException();
     }
 
+    void validateWritePreference(FailureCollector failureCollector, String writePreference) {
+      Set<String> validPreferences = ImmutableSet.of(
+          JobInfo.WriteDisposition.WRITE_EMPTY.name(),
+          JobInfo.WriteDisposition.WRITE_APPEND.name(),
+          JobInfo.WriteDisposition.WRITE_TRUNCATE.name()
+      );
+      if (!validPreferences.contains(writePreference)) {
+        failureCollector.addFailure(
+            String.format("Invalid write preference '%s'. Allowed values are '%s', '%s', or '%s'.",
+                writePreference,
+                JobInfo.WriteDisposition.WRITE_EMPTY.name(),
+                JobInfo.WriteDisposition.WRITE_APPEND.name(),
+                JobInfo.WriteDisposition.WRITE_TRUNCATE.name()
+            ),
+            "Please provide a valid write preference."
+        )
```

Comment:
you can just do `Invalid write preference '%s'. Allowed values are validPreferences.toString()`